sdgym/datasets.py

```diff
@@ -69,9 +86,11 @@ def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_sec
     return metadata
 
 
-def load_tables(metadata):
+def load_tables(metadata, max_rows=None):
     real_data = metadata.load_tables()
     for table_name, table in real_data.items():
+        if max_rows and table.shape[0] > max_rows:
```

Comment:
You can skip this check and directly get `table.head(max_rows)`, since `table.head(None) == table`, and `table.head(len(table) + K > 0) == table`